ff/src/biginteger/macros.rs

```diff
@@ -352,5 +352,48 @@ macro_rules! bigint_impl {
                 repr
             }
         }
+
+        impl From<BigUint> for $name {
+            #[inline]
+            fn from(val: BigUint) -> $name {
+                use num_traits::ToPrimitive;
+                use num_traits::Zero;
+
+                let mut tmp = val.clone();
+                let mut tmp2 = val.clone();
+
+                let mut limbs = [0u64; $num_limbs];
+
+                for i in 0..$num_limbs {
+                    tmp >>= 64;
+                    tmp <<= 64;
+
+                    limbs[i] = (tmp2 - tmp.clone()).to_u64().unwrap();
+
+                    tmp >>= 64;
+                    tmp2 = tmp.clone();
```

Comment:
I see how this works, but it seems to have quadratic cost in the number of limbs. Is there no better way?

Reply:
You are right. Maybe I should use a better way: BigUint has `from_bytes_le` and `to_bytes_le`, from which I may have a better construction.